toolchain/base/value_store.h

```diff
@@ -221,62 +221,67 @@ auto CanonicalValueStore<IdT>::Reserve(size_t size) -> void {
   values_.Reserve(size);
 }
 
-// A ValueStore keyed on `FromIdT`, and maps that to a `StoredIdT`.
-template <typename FromIdT, typename StoredIdT>
+// A ValueStore that builds a 1:1 relationship between two IDs.
+// * `RelatedIdT` represents a related ID that can be used to find values in the
+//   store.
+// * `IdT` is the actual ID of values in this store, and `IdT::ValueType` is the
+//   value type being stored.
+//
+// The value store builds a mapping so that either ID can be used later to find
+// a value. And the user can query if a related `RelatedIdT` has been used to
+// add a value to the store or not.
+//
+// When adding to the store, the user provides the related `RelatedIdT` along
+// with the value being stored, and gets back the ID of the value in the store.
```

Comment:
Perhaps mention that the index values are the same, and that this will allocate memory for all ids (up to the maximum related id added), even those that aren't used?

Reply:
Done